lib/Authentication/AuthTokens/ShibAuthToken.php

```diff
@@ -204,4 +204,27 @@ public static function isStateless() {
         return true;
     }
 
+    private function getEntitlementErrorMessage()
+    {
+        $refedsResAndSchURL = "https://refeds.org/category/research-and-scholarship";
+        $refedsSirtfiURL = "https://refeds.org/sirtfi";
+        $resourceLink = "https://docs.egi.eu/internal/configuration-database/access";
+        $sectionFragmentInfo = "/#using-institutional-account-via-egi-check-in";
+        $documentationURL = $resourceLink . $sectionFragmentInfo;
+
+        return '<ul>'
+            . '<li>Login requires the entitlement '
+            . 'urn:mace:egi.eu:res:gocdb#aai.egi.eu, '
+            . 'which was not provided. This entitlement is automatically '
+            . 'granted when using an identity provider compliant with '
+            . "<a href=\"{$refedsResAndSchURL}\" target='_blank'>"
+            . "REFEDS R&amp;S</a> and "
+            . "<a href=\"{$refedsSirtfiURL}\" target='_blank'>"
+            . "REFEDS Sirtfi</a>. "
+            . 'Please see here for more information: '
+            . '<a href="' . $documentationURL . 'target="_blank">'
+            . $documentationURL . '</a>.'
+            . '</li>'
+            . '</ul>';
```

Comment:
Please split this so the individual sentences are on their own bullet point:
- Login requires...
- This entitlement...
- Please see here...

Reply:
Sure Greg, I have updated the code to reflect this proposal in https://github.com/GOCDB/gocdb/pull/518/commits/b9c9dd9e17c1c5da9f99b65eb71b589346566194